### PR TITLE
GoogleMapsOverlay support non-interleaved mode for vector maps

### PR DESCRIPTION
**Pull request by felixpalmer**
For https://github.com/visgl/deck.gl/issues/6296

#### Change List
- Add `interleaved` prop to `GoogleMapsOverlay` (default `false` to provide back-compatibility)
- Implement rendering path for vector maps on separate overlaid canvas
- Test updates

### CODE REVIEW

File: modules/google-maps/src/google-maps-overlay.js [hunk 1]
```diff
@@ -81,6 +84,7 @@ export default class GoogleMapsOverlay {
 
   /* Private API */
   _createOverlay(map) {
+    const {interleaved} = this.props;
```
Comment (Pessimistress) [inline modules/google-maps/src/google-maps-overlay.js:87]:
```suggestion
    const {interleaved = true} = this.props;
```

And you can drop the mutation in the constructor.

File: modules/google-maps/src/utils.js [hunk 2]
```diff
@@ -204,9 +204,8 @@ export function getViewPropsFromCoordinateTransformer(map, transformer) {
   const focalDistance = 0.5 * projectionMatrix[5];
 
   return {
-    // Using external gl context - do not set css size
-    width: false,
-    height: false,
```
Comment (Pessimistress) [inline modules/google-maps/src/utils.js:209]: Why remove this? It prevents Deck from scaling the gl context size to device pixel ratio. Removing ii=t will break context sharing.

**Comment by zakjan** [comment]
Thanks, I can confirm this works.

`pointerEvents: 'none'` is unfortunate (because it prevents mouse events and tooltips from working), but it's in the Mapbox overlay as well, so I guess it's acceptable

**Comment by felixpalmer** [comment]
> Thanks, I can confirm this works.
> 
> `pointerEvents: 'none'` is unfortunate (because it prevents mouse events and tooltips from working), but it's in the Mapbox overlay as well, so I guess it's acceptable

I tested it and mouse events do work, as we proxy them: https://github.com/visgl/deck.gl/blob/f0649b69d6a91e0aec146da9c6d30d60ccf4d1ef/modules/google-maps/src/utils.js#L252-L289

Did you come across something that was broken?

File: modules/google-maps/src/google-maps-overlay.js (hunk 1 above)
Comment (felixpalmer) [reply modules/google-maps/src/google-maps-overlay.js:87]: As I also read the value out in utils.js I wanted it to be defaulted in one place. I've removed the mutation.

**Review by Pessimistress — APPROVED** [review]

**Comment by zakjan** [comment]
> Did you come across something that was broken?

Ok, sorry my mistake, the canvas I tested it with was overlayed by another element. Events works :+1:

**Pull request merged**